nexus-context-propagation/interceptor.go

```diff
@@ -0,0 +1,114 @@
+package nexuscontextpropagation
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"github.com/temporalio/samples-go/ctxpropagation"
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/workflow"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type WorkerInterceptor struct {
```

Comment:
I am doing this in Java right now and one thing I debated was just putting all the context propagation in the interceptor. Is is generally what we recommend to customers now

Reply:
Why do we recommend that? I wouldn't expect users to have to know every method that context might be propagated from. I think newer SDKs should implement a context propagation interceptor that could potentially include Nexus if we have a good way to ensure a consistent format (e.g. everything is a payload that can be encrypted).

Reply:
> SDKs could provide a context propagation interceptor helper like we do for tracing

Definitely agree with this. While it's simple to implement one, you still need to know which contexts to propagate to and from and remember to update your interceptors for new APIs that require interception.